edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -3851,11 +3851,7 @@ private void batchPersistEntities(final List<?> entities) {
 		final Session session = getSessionFactory().openSession();
 
 		for (int i = 0; i < entities.size(); i++) {
-			session.saveOrUpdate(entities.get(i));
-			if (i % 20 == 0) {
-				session.flush();
-				session.clear();
-			}
+			session.update(entities.get(i));
```

Comment:
question for @steveswinsburg : was session.flush() added for a specific reason/use case in 503c00165e6?